pkg/controllers/provisioning/suite_test.go

```diff
@@ -113,6 +115,77 @@ var _ = Describe("Provisioning", func() {
 		Expect(len(nodes.Items)).To(Equal(1))
 		ExpectScheduled(ctx, env.Client, pod)
 	})
+	It("should provision single pod if no other pod is received within the batch idle duration", func() {
+		ExpectApplied(ctx, env.Client, test.NodePool())
+		pod := test.UnschedulablePod()
+		ExpectApplied(ctx, env.Client, pod)
+		prov.Trigger(pod.UID)
+
+		wg := sync.WaitGroup{}
+		ExpectToWait(fakeClock, &wg)
+		result := ExpectSingletonReconciled(ctx, prov)
+		Expect(result.RequeueAfter).ToNot(BeNil())
+	})
+	It("should not extend the timeout if we receive the same pod within the batch idle duration", func() {
+		ctx = options.ToContext(ctx, test.Options(test.OptionsFields{
+			BatchMaxDuration:  lo.ToPtr(10 * time.Second),
+			BatchIdleDuration: lo.ToPtr(5 * time.Second),
+		}))
+		ExpectApplied(ctx, env.Client, test.NodePool())
+		pod := test.UnschedulablePod()
+		ExpectApplied(ctx, env.Client, pod)
+		prov.Trigger(pod.UID)
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		Expect(fakeClock.HasWaiters()).To(BeFalse())
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			fakeClock.Step(3 * time.Second)
```

Comment:
nit: Consider adding comments to help someone understand why you are stepping the times that you are and why you are checking on waiters, etc.

This should improve the readability for someone coming back to it in the future